tests/s3_util_tests.c

```diff
@@ -364,6 +364,151 @@ static int s_test_s3_get_num_parts_and_get_part_range(struct aws_allocator *allo
+
+    const struct s3_request_part_config_example valid_request_part_config[] = {
+        {
+            .name = "simple case",
+            .content_length = 10000,
+            .client_part_size = 5,
+            .client_max_part_size = default_max_part_size,
+            .expected_part_size = 5,
+            .expected_num_parts = 2000,
+        },
+        {
+            .name = "large content length with small part size",
+            .content_length = 990000,
+            .client_part_size = 5,
+            .client_max_part_size = default_max_part_size,
+            .expected_part_size = 99,
+            .expected_num_parts = 10000,
+        },
+        {
+            .name = "large content length with large part size",
+            .content_length = 1000000,
+            .client_part_size = 500,
+            .client_max_part_size = default_max_part_size,
+            .expected_part_size = 500,
+            .expected_num_parts = 2000,
+        },
+        {
+            .name = "large odd content length",
+            .content_length = 995649,
+            .client_part_size = 5,
+            .client_max_part_size = default_max_part_size,
+            .expected_part_size = 100,
+            .expected_num_parts = 9957,
+        },
+        {
+            .name = "10k",
+            .content_length = 50000,
+            .client_part_size = 5,
+            .client_max_part_size = default_max_part_size,
+            .expected_part_size = 5,
+            .expected_num_parts = 10000,
+        },
+        {
+            .name = "10k - 1 parts",
+            .content_length = 49995,
+            .client_part_size = 5,
+            .client_max_part_size = default_max_part_size,
+            .expected_part_size = 5,
+            .expected_num_parts = 9999,
+        },
+        {
+            .name = "10k with small last part",
+            .content_length = 49998,
+            .client_part_size = 5,
+            .client_max_part_size = default_max_part_size,
+            .expected_part_size = 5,
+            .expected_num_parts = 10000,
+        },
+        {
+            .name = "10k + 1 parts",
+            .content_length = 50001,
+            .client_part_size = 5,
+            .client_max_part_size = default_max_part_size,
+            .expected_part_size = 6,
+            .expected_num_parts = 8334,
+
+        },
+        {
+            .name = "bump content length",
+            .content_length = 100000,
+            .client_part_size = 5,
+            .client_max_part_size = default_max_part_size,
+            .expected_part_size = 10,
+            .expected_num_parts = 10000,
+        },
+        {
+            .name = "bump content length with non-zero mod",
+            .content_length = 999999,
+            .client_part_size = 5,
+            .client_max_part_size = default_max_part_size,
+            .expected_part_size = 100,
+            .expected_num_parts = 10000,
+        },
+        {
+            .name = "5 tb content length",
```

Comment:
the numbers in real life is based on Bytes, instead of MB.

Reply:
Thanks, fixed.